packages/java-parser/src/productions/classes.js

```diff
@@ -359,11 +359,23 @@ function defineRules($, t) {
   $.RULE("methodDeclarator", () => {
     $.CONSUME(t.Identifier);
     $.CONSUME(t.LBrace);
-    $.OPTION(() => {
+    $.OPTION({
+      // a "formalParameterList" and a "receiverParameter"
+      // cannot be distinguished using fixed lookahead.
+      GATE: $.BACKTRACK($.receiverParameter),
+      DEF: () => {
+        $.SUBRULE($.receiverParameter);
+        $.OPTION2({
+          GATE: () => $.LA(1).tokenType !== t.RBrace,
```

Comment:
Is it really optional ? It does not seems to be possible according to the spec: https://docs.oracle.com/javase/specs/jls/se21/html/jls-8.html#jls-8.4.1-220